lib/modules/webserver/server.js

```diff
@@ -24,13 +33,26 @@ class Server {
       serve(req, res, finalhandler(req, res));
     }).withShutdown();
 
+    if (isFirstStart) {
+      const html = embark_building_placeholder({buildingMsg: __('Embark is building, please wait...')});
+      fs.mkdirpSync(this.dist); // create dist/ folder if not already exists
+      fs.writeFileSync(utils.joinPath(this.dist, 'index.html'), html);
```

Comment:
Isn't this file already created by the pipeline? Wouldn't this just write it a second time for no reason?

Reply:
Currently, if a browser is opened as soon as soon as the webserver is ready (on first build or after `embark reset`), an error-text will be displayed as the page in the browser because there is nothing to serve yet — pipeline hasn't written the placeholder page into `buildDir` yet. You can try it on develop or 3.1.x to see the behavior, it's always kind of bugged me. That error page doesn't auto-refresh, so it seems like a good idea to put the auto-refreshing page into place.

My first attempt at this was to put the `opn` logic in the pipeline, but there are a couple of difficulties. For one, the webserver config would need to be exposed to the pipeline (for host and port). Also, the pipeline doesn't differentiate between `embark run` and `embark build`, so a browser is opened pointlessly during `embark build` (I tried it); we would need to revise pipeline so it's aware whether the context is `run` or `build`.

Given all the above, I think the duplication is okay, but I'm open to suggestions.

Reply:
Once we have a webpack dev server, it will work differently, so this would really just be a temporary but convenient feature.

Reply:
Could you put an event in the pipeline then that when called, creates the index file. That would reduce the duplication.

Reply:
Okay, I gave it some thought and moved the responsibility of building the placeholder to the webserver module, along with supporting changes.